docs/migration/v4.md

```diff
@@ -0,0 +1,30 @@
+# v4 Migration Guide
+
+This document describes the changes made in `v4` of **NCronJob** and how to migrate from `v3`.
+
+Version 4 of **NCronJob** introduces some breaking changes to improve the API.
+
+## `UseNCronJob` and `UseNCronJobAsync` methods
+Following the classical ASP.NET pattern, the `UseNCronJob` and `UseNCronJobAsync` methods have been introduced to the `IHost` interface. The minimal setup now looks like this:
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+builder.Services.AddNCronJob(...);
+
+var app = builder.Build();
+await app.UseNCronJobAsync();
+app.Run();
+```
+
+Instead of the async method `UseNCronJobAsync`, you can also use the synchronous `UseNCronJob` method. The difference is that the async method returns a `Task` that you can await.
+
+## `RunAtStartup` method
+The `RunAtStartup` method has a bit different semantics than before. All startup jobs are now executed when `UseNCronJob` or `UseNCronJobAsync` is called. This ensures that all startup jobs are completed before any other part of the application is executed.
+
+## `AddNCronJob(Delegate)` move to `NCronJobExtensions`
```

Comment:
I'm not sure that changes anything from a consumer standpoint. 

Are you sure this belongs to the migration guide, rather than the changelog?

Reply:
Well, we are blocking maybe API requests because there is a job still running in the background.
To get back the old behavior, the user has to actively "ignore" the task that represents `USeNCronJobAsync`. 

That is somewhat of a breaking change to me.

Reply:
My comment was related to "## `AddNCronJob(Delegate)` move to `NCronJobExtensions`" :wink:

Reply:
Ahhhhh sorry - you got me there. I shouldn't do multiple things at the same time.

So on the rare occasion someone used the extension like this:
```csharp
IServiceCollectionExtensions.AddNCronJob(() => {}, "* * * * * ");
```

Instead of the extension-way, this would lead to a compiler error as I moved the stuff to `NCronJobExtensions`.
So from an API point-of-view that is indeed a breaking change.

Reply:
Wow....

I actually never realized one could leverage extension methods this way.

Makes sense then 

(Although I'm curious about the use cases where one would need to use this syntax :D)